scheduler/data-flow/src/test/kotlin/io/seldon/dataflow/kafka/PipelineStatusTest.kt

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright (c) 2024 Seldon Technologies Ltd.
+
+Use of this software is governed BY
+(1) the license included in the LICENSE file or
+(2) if the license included in the LICENSE file is the Business Source License 1.1,
+the Change License after the Change Date as each is defined in accordance with the LICENSE file.
+*/
+
+package io.seldon.dataflow.kafka
+
+import org.apache.kafka.streams.KafkaStreams
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.Arguments.arguments
+import org.junit.jupiter.params.provider.MethodSource
+import strikt.api.expect
+import strikt.assertions.isEqualTo
+import java.util.stream.Stream
+
+internal class PipelineStatusTest {
+    @DisplayName("Test PipelineStatus States")
+    @ParameterizedTest(name = "{0} expected isActive:{1} | isError:{2}")
+    @MethodSource()
+    fun checkState(
+        testName: String,
+        expectedIsActive: Boolean,
+        expectedIsError: Boolean,
+        state: PipelineStatus,
+    ) {
+        val statusIsActive = state.isActive()
+        val statusIsError = state.isError()
+
+        expect {
+            that(statusIsActive).isEqualTo(expectedIsActive)
+            that(statusIsError).isEqualTo(expectedIsError)
+        }
+    }
+
+    companion object {
+        private const val IS_ACTIVE = true
+        private const val IS_ERROR = true
+
+        @JvmStatic
+        fun checkState(): Stream<Arguments> =
```

Comment:
is there a way to parametrize choices as opposed to listing them?

Reply:
The test is already parameterised (it is written once in the checkState function above -- the one outside the companion object). The following lists the states and the expected results for the given state. If one were to automate the way the expected results are created then you would need to test that the automation produces the expected expected results and so on... So I think further parameterising here would not help, but perhaps I'm misunderstanding your point

Reply:
After our discussion: it would be indeed worthwhile to think of ways of ensuring that all possible combinations have been covered; Those type of combinatorial tests are possible, but not for the case where the expected values need to be manually provided per test case.